fetcher-v2/admin/node.go

```diff
@@ -0,0 +1,166 @@
+package admin
+
+import (
+	"strconv"
+	"time"
+
+	"bisonai.com/fetcher/v2/utils"
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/log"
+)
+
+func discover(c *fiber.Ctx) error {
+	h, err := utils.GetHost(c)
+	if err != nil {
+		log.Panicf("failed to load host: %s", err)
+	}
+
+	s, err := utils.GetDiscoverString(c)
+	if err != nil {
+		log.Panicf("failed to load discover string: %s", err)
+	}
+
+	go utils.DiscoverPeers(c.Context(), *h, s)
+
+	return c.SendString("triggered peer discovery")
+}
+
+func getAllNodesInfo(c *fiber.Ctx) error {
+	nodes, err := utils.GetNodes(c)
+	if err != nil {
+		log.Panicf("failed to load nodes: %s", err)
+	}
+
+	info := make([]string, 0, len(nodes))
+	// Iterate over the nodes and call PrintInfo on each one
+	for _, node := range nodes {
+		info = append(info, node.String())
+	}
+
+	// Return the info slice as the JSON response
+	return c.JSON(info)
+}
+
+func getNodeInfo(c *fiber.Ctx) error {
+	topicString := c.Params("topic")
+	node, err := utils.GetNode(c, topicString)
+	if err != nil {
+		log.Panicf("failed to load node: %s", err)
+	}
+
+	return c.JSON(node.String())
+}
+
+func addNode(c *fiber.Ctx) error {
+	h, err := utils.GetHost(c)
+	if err != nil {
+		log.Panicf("failed to load host: %s", err)
+	}
+
+	topicString := c.Params("topic")
+	node, err := utils.NewNode(c.Context(), *h, topicString)
+	if err != nil {
+		log.Panicf("failed to create node: %s", err)
+	}
+
+	err = utils.SetNode(c, topicString, node)
+	if err != nil {
+		log.Panicf("failed to set node: %s", err)
+	}
+
+	nodes, err := utils.GetNodes(c)
+	if err != nil {
+		log.Panicf("failed to load nodes: %s", err)
+	}
+
+	info := make([]string, 0, len(nodes))
+	// Iterate over the nodes and call PrintInfo on each one
+	for _, node := range nodes {
+		info = append(info, node.String())
+	}
+
+	// Return the info slice as the JSON response
+	return c.JSON(info)
+}
+
+func startNode(c *fiber.Ctx) error {
+	topicString := c.Params("topic")
```

Comment:
We should probably make a check on whether the `topicString` is an empty string. The definition of parameter is in `topic` name parameter is in `admin.go` which could cause issue later when somebody decides to update it. Morever, we could consider sending it as data payload since it is a POST request.